.github/workflows/pulsar-ci.yaml

```diff
@@ -127,11 +127,11 @@ jobs:
           action: wait
 
   macos-build:
-    name:
+    name: Build Pulsar on MacOS
     runs-on: macos-latest
     timeout-minutes: 120
     needs: [ 'changed_files_job', 'build-and-license-check' ]
```

Comment:
This should be moved to late stages of the pipeline so that it happens after integration tests. The reason for this is that there will be less re-runs at that stage. Our current assumption of the CI queuing problem is that when a workflow gets re-run, all previously executed jobs get counted in the usage reporting as if they were run again.

Reply:
Right. Moved at the same "step" of `system-tests`